crates/server/src/rpc/service.rs

```diff
@@ -175,15 +182,17 @@ impl QuestsServiceServer<QuestsRpcServerContext, UnableToOpenStream>
                         let generator_yielder = yielder.clone();
                         let ids = quest_instance_ids.clone();
                         async move {
-                            if let Some(user_update::Message::QuestState(state)) = &user_update.message {
-                                if ids.contains(&state.quest_instance_id) {
-                                    if generator_yielder.r#yield(user_update).await.is_err() {
-                                        error!(
-                                            "User Update received > Couldn't send update to subscriptors"
-                                        );
-                                        return false // Just return false on failure
-                                    } else {
-                                        return true
+                            if let Some(user_update::Message::QuestStateUpdate(state)) = &user_update.message {
```

Comment:
Is it possible that we may be interested in other message variants too?

Reply:
Not currently. It only receives that variant in this callback.